src/add_data_advanced.py

```diff
@@ -9,7 +9,7 @@ def open_add_data_advanced_window(widget, _, self):
     """
     win = AddAdvancedWindow(self)
     button = win.open_advanced_confirm_button
-    win.set_transient_for(self.props.active_window)
+    win.set_transient_for(self.main_window)
```

Comment:
Yes, this is a good change. In the first implimentation, I just called props_active_window seperately, but using the static self.main_window is more robust.